library_generation/owlbot/synthtool/languages/java.py

```diff
@@ -0,0 +1,691 @@
+
+        metadata["latest_version"] = latest_maven_version(
+            group_id=group_id, artifact_id=artifact_id
+        )
+
+    metadata["latest_bom_version"] = latest_maven_version(
+        group_id="com.google.cloud",
+        artifact_id="libraries-bom",
+    )
+
+    metadata["samples"] = samples.all_samples(["samples/**/src/main/java/**/*.java"])
+    metadata["snippets"] = snippets.all_snippets(
+        ["samples/**/src/main/java/**/*.java", "samples/**/pom.xml"]
+    )
+    if repo_metadata and "min_java_version" in repo_metadata:
+        metadata["min_java_version"] = repo_metadata["min_java_version"]
+    else:
+        metadata["min_java_version"] = DEFAULT_MIN_SUPPORTED_JAVA_VERSION
+
+    return metadata
+
+
+def common_templates(
+    excludes: List[str] = [],
+    template_path: Optional[Path] = None,
+    **kwargs,
+) -> None:
+    """Generate common templates for a Java Library
+
+    Fetches information about the repository from the .repo-metadata.json file,
+    information about the latest artifact versions and copies the files into
+    their expected location.
+
+    Args:
+        :param excludes: List of template paths to ignore
+        :param template_path:
+        :param kwargs: Additional options for CommonTemplates.java_library()
+    """
+    metadata = _common_template_metadata()
+    kwargs[METADATA] = metadata
+
+    # Generate flat to tell this repository is a split repo that have migrated
+    # to monorepo. The owlbot.py in the monorepo sets monorepo=True.
+    monorepo = kwargs.get("monorepo", False)
+    kwargs["monorepo"] = monorepo
+    split_repo = not monorepo
+    repo_metadata = metadata["repo"]
+    repo_short = repo_metadata["repo_short"]
+    if os.getenv(LIBRARIES_BOM_VERSION_ENV_KEY, default=None) is not None:
+        kwargs[METADATA][LIBRARIES_BOM_VERSION] = os.getenv(
+            LIBRARIES_BOM_VERSION_ENV_KEY
+        )
+    kwargs[METADATA][LIBRARY_VERSION] = os.getenv(LIBRARY_VERSION_ENV_KEY)
+    # Special libraries that are not GAPIC_AUTO but in the monorepo
+    special_libs_in_monorepo = [
+        "java-translate",
+        "java-dns",
+        "java-notification",
+        "java-resourcemanager",
+    ]
+    kwargs["migrated_split_repo"] = split_repo and (
+        repo_metadata["library_type"] == "GAPIC_AUTO"
+        or (repo_short and repo_short in special_libs_in_monorepo)
+    )
+    logger.info(
+        "monorepo: {}, split_repo: {}, library_type: {},"
+        " repo_short: {}, migrated_split_repo: {}".format(
+            monorepo,
+            split_repo,
+            repo_metadata["library_type"],
+            repo_short,
+            kwargs["migrated_split_repo"],
+        )
+    )
+
+    templates = gcp.common.CommonTemplates(template_path=template_path).java_library(
+        **kwargs
+    )
+
+    # skip README generation on Kokoro (autosynth)
+    if os.environ.get("KOKORO_ROOT") is not None:
```

Comment:
I think we can remove this line since we won't run tests/build in Kokoro.

Reply:
Good point. Removed